File: 5week/남규/4. Evaluate Reverse Polish Notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<string> stk;
        
        for(auto token : tokens){
            if((token == "+") || (token == "-") || (token == "*") || (token == "/")){
                int operand2 = stoi(stk.top());
                stk.pop();
                int operand1 = stoi(stk.top());
                stk.pop();
                
                if(token == "+"){
                    stk.push(to_string(operand1 + operand2));
                } else if(token == "-"){
                    stk.push(to_string(operand1 - operand2));
                } else if(token == "*"){
                    stk.push(to_string(operand1 * operand2));
                } else if(token == "/"){
                    stk.push(to_string(operand1 / operand2));
                }
            } else {
                stk.push(token);
            }
        }
        return stoi(stk.top());
    }
};

File: 5week/근우/4.evaluate-reverse-polish-notation.py
# ## https://leetcode.com/problems/evaluate-reverse-polish-notation/
# from typing import List


from typing import List


class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        nums = []
        operator = set({"+", "*", "/", "-"})
        for token in tokens:
            if token in operator:
                num1, num2 = nums.pop(), nums.pop()
                if token == "+":
                    nums.append(num2 + num1)
                elif token == "*":
                    nums.append(int(num2 * num1))
                elif token == "-":
                    nums.append(num2 - num1)
                else:
                    nums.append(int(num2 / num1))
            else:
                nums.append(int(token))
        return nums[0]
